daemons/fenced/fenced_remote.c

```diff
@@ -2176,6 +2227,83 @@ process_remote_stonith_query(xmlNode * msg)
     return pcmk_ok;
 }
 
+static gboolean
+check_async_reply_cb(gpointer data)
+{
+    check_async_reply_t *a = data;
+    a->timer = 0;
+
+    crm_trace("Asynchronous response wait timeout.(op=%s)", a->remote_op_id);
+
+    if (stonith_remote_op_list) {
+        remote_fencing_op_t *remote_op = NULL;
+
+        remote_op = g_hash_table_lookup(stonith_remote_op_list, a->remote_op_id);
+        if (remote_op != NULL) {
+            /* The op->result set is not needed as it is already set with fenced_process_fencing_reply(). */
+            remote_op->state = st_failed;
+            /* Set the time when it actually failed. */
+            remote_op->completed = a->completed;
+            remote_op->completed_nsec = a->completed_nsec;
+
+            /* Set local operation information to failure. */
+            pcmk__xe_set_bool_attr(a->msg, F_STONITH_SET_COMPLETED, false);
```

Comment:
I don't think we need to add a new attribute to the message ... finalize_op() could set the completed time only if it's zero

Reply:
Okay!